.github/workflows/test.yaml

```diff
@@ -0,0 +1,45 @@
+name: test
```

Comment:
I think it would be better to merge this WF with the "build" WF, the build step takes alot of time and rebuilding here again for the test seems redundant to me. what I'm suggesting is to just take the `make test` step from here and move it into the `build` workflow (which would be renamed to build-test)

Reply:
Sure! Addressing this in the next commit.